server/activate_hooks.go

```diff
@@ -3,37 +3,54 @@ package main
 import (
 	"fmt"
 
+	"github.com/blang/semver"
 	"github.com/mattermost/mattermost-server/model"
 )
 
+const minimumServerVersion = "5.4.0"
+
 // OnActivate is invoked when the plugin is activated.
 //
 // This demo implementation logs a message to the demo channel whenever the plugin is activated.
 func (p *Plugin) OnActivate() error {
-	teams, err := p.API.GetTeams()
+	serverVersion, err := semver.Parse(p.API.GetServerVersion())
 	if err != nil {
 		p.API.LogError(
-			"failed to query teams OnActivate",
+			"failed to parse server version",
 			"error", err.Error(),
 		)
+		return err
+	}
+	r := semver.MustParseRange(">=" + minimumServerVersion)
+	if !r(serverVersion) {
+		return fmt.Errorf("current Mattermost version is to low. Please update your Mattermost Server to a least v%s.", minimumServerVersion)
+	}
+
+	teams, appErr := p.API.GetTeams()
```

Comment:
can you revert this `appErr` change as well? (just leave it named `err`)

Reply:
Oh, my mistake! I didn't clue into the fact that you had already declared `err` as an `error`, in which case `appError` is perfectly fine. I thought it was just an artifact of the other refactoring not undone.

The medium-term plan is to move towards the `error` interface and only deal with `AppError` when we need the fields it exposes. Otherwise, it's more idiomatic to use `error` everywhere, to avoid the kind of pain you're seeing here.